packages/calcite-components/src/components/block-section/block-section.tsx

```diff
@@ -199,21 +211,35 @@ export class BlockSection implements LocalizedComponent, T9nComponent, LoadableC
   }
 
   render(): VNode {
-    const { el, messages, open, text, toggleDisplay } = this;
-    const dir = getElementDir(el);
-    const arrowIcon = open
-      ? ICONS.menuOpen
-      : dir === "rtl"
-        ? ICONS.menuClosedLeft
-        : ICONS.menuClosedRight;
+    const { messages, open, text, toggleDisplay, iconFlipRtl } = this;
+    const arrowIcon = open ? ICONS.menuOpen : ICONS.menuClosed;
+
+    const iconStartEl = this.iconStart ? (
+      <calcite-icon
```

Comment:
To simplify a bit, can you look at adding a `renderIcon` method that returns an icon with the corresponding start/end props?